### DISCUSSION TITLE
Custom error pages with livewire components

### DISCUSSION THREAD

**Discussion by wizzymore** [livewire/livewire#8593] [7. Bugs (archived)]
### Livewire version

v3.5.1

### Laravel version

v11.11.1

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Create a Route::fallback() that renders a page with a livewire component.
Go to a 404 route and see that the livewire component works.

Go to a route that fails the SubstitudeBindings middleware and see that the livewire component doesn't work and shows a 404 page modal.

Step by step for the repository provided:  
Clone the repo, run `php artisan migrate:fresh --seed`.
Go to /1 and see a white page ( this means the user was fetched )
Go to /2 and see a 404 error page
Click the "testing" button to see the 404 livewire modal because SubstituteBindings fails (BUG)

Comment the "/{user}" route from web.php
Do that again by visiting either /1, /2 or /whatever and click the testing button. Now a javascript alert is properly triggered by livewire.

### Repository or Wirebox URL

https://github.com/wizzymore/livewire-404-problem

### Additional information

This happens because the PersistentMiddleware filters everything except the substitude bindings middleware, which i don't think it's correct. Why would laravel care if the page it is on has substituted the bindings or not? I want the component to render everywhere i put it, and i want updates to work wherever i have a valid CSRF TOKEN.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by utsavsomaiya** [comment]
@joshhanley and @calebporzio 

When in line livewire component is used like logout on custom error page it will open another window and display 404 message!

**Comment by utsavsomaiya** [comment]
Event I used custom route to set same as documentation : https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint

**Comment by nox-4O4** [comment]
Have a look at https://github.com/livewire/livewire/discussions/7511, the workaround described in https://github.com/livewire/livewire/discussions/7511#discussioncomment-12273665 is also applicable here.

> Why would laravel care if the page it is on has substituted the bindings or not? I want the component to render everywhere i put it, and i want updates to work wherever i have a valid CSRF TOKEN.

Sending the request again through the middleware stack is required to ensure the update is authenticated. As the `Authorize` middleware might depend on resolved models it is necessary to resolve bindings first.
